### PR TITLE
fix: support BYTE-LENGTH clause for utf-8 pic clause #2558

### PR DESCRIPTION
**Pull request by amanPras**
## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test #2558
- [x] Test  BYTE-LENGTH phrase can only be specified when the picture string consists of a single ‘U’ symbol 
      refer: https://www.ibm.com/docs/en/cobol-zos/6.4?topic=entry-picture-clause

## Checklist:
- [x] Each of my commits contains one meaningful change
- [x] I have performed rebase of my branch on top of the development
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: server/parser/src/main/antlr4/org/eclipse/lsp/cobol/core/CobolLexer.g4 [hunk 1]
```diff
@@ -918,8 +918,10 @@ COMMASEPARATOR : ', '  {enableCobolSpecialSeparators}? -> channel(HIDDEN);
 // PICTURECLAUSE mode
 // -------------------------------------------------------
 mode PICTURECLAUSE;
-FINALCHARSTRING: CHARSTRING+ ->popMode;
+FINALCHARSTRING: (CHARSTRING+ | SINGLE_UTF_CHARSTRING) ->popMode;
```
Comment (amanPras) [inline server/parser/src/main/antlr4/org/eclipse/lsp/cobol/core/CobolLexer.g4]: I am not sure this approach is right or should we deal with it [here](https://github.com/eclipse-che4z/che-che4z-lsp-for-cobol/blob/f8be5ea66b8e4661ff1827886bc02869232036c1/server/parser/src/main/antlr4/org/eclipse/lsp/cobol/core/CobolParser.g4#L662)

Comment (amanPras) [reply server/parser/src/main/antlr4/org/eclipse/lsp/cobol/core/CobolLexer.g4]: This obviously, saves some validation processor classes and BYTE-LENGTH lexeme collision with other rules like cobolWord

Comment (slavek-kucera) [reply server/parser/src/main/antlr4/org/eclipse/lsp/cobol/core/CobolLexer.g4]: This approach will make it harder to extract the number if it is ever needed.
I'd suggest trying to ban the single `U` from FINALCHARSTRING, then redefined `charString` as `FINALCHARSTRING | U` and modify the `dataPictureClause` rule as `U (BYTE_LENGTH PICTUREIS? INTEGERLITERAL)? | charString`. Not sure what the purpose of `+` is in that rule.

**Review by slavek-kucera — APPROVED** [review]

**Comment by amanPras** [comment]
#2558

File: server/parser/src/main/antlr4/org/eclipse/lsp/cobol/core/CobolLexer.g4 [hunk 2]
```diff
@@ -920,6 +920,8 @@ COMMASEPARATOR : ', '  {enableCobolSpecialSeparators}? -> channel(HIDDEN);
 mode PICTURECLAUSE;
 FINALCHARSTRING: CHARSTRING+ ->popMode;
 DOT_FS2 : '.' EOF? -> popMode;
+SINGLE_U_CHAR_BYTE_LENGTH: U WS2+ BYTE_LENGTH_P ->popMode;
```
Comment (slavek-kucera) [inline server/parser/src/main/antlr4/org/eclipse/lsp/cobol/core/CobolLexer.g4]: Should this include newlines?

Comment (amanPras) [reply server/parser/src/main/antlr4/org/eclipse/lsp/cobol/core/CobolLexer.g4]: updated

**Review by slavek-kucera — APPROVED** [review]

**Pull request merged**